### PR TITLE
Document why auth providers aren't validated in the admin API

### PR DESCRIPTION
**Pull request by clokep**
This updates some docstrings with some info from discussions

Open questions:

* Does this sound fine or should we really be validating those?
* Should we update the markdown documentation?
* Should we update the documentation for the module API?

### CODE REVIEW

File: changelog.d/12004.doc [hunk 1]
```diff
@@ -0,0 +1 @@
+Clarify information about external IdP IDs.
```
Comment (reivilibre) [inline changelog.d/12004.doc:1]:
```suggestion
Clarify information about external Identity Provider IDs.
```
if that's right?

File: synapse/storage/databases/main/registration.py [hunk 2]
```diff
@@ -622,10 +622,13 @@ async def record_user_external_id(
     ) -> None:
         """Record a mapping from an external user id to a mxid
 
+        See notes in _record_user_external_id_txn about what constitutes validate data.
```
[the remark this answers is not in the document]
Comment (reivilibre) [reply synapse/storage/databases/main/registration.py:625]: note it's present also below

**Review by reivilibre — APPROVED** [review]

File: synapse/storage/databases/main/registration.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (clokep) [reply synapse/storage/databases/main/registration.py:625]: Yes, it was a typo. Should say "valid data". 👍

**Comment by clokep** [comment]
@reivilibre / @babolivier Do you have any thoughts on the questions in the description?

**Comment by babolivier** [comment]
> Does this sound fine or should we really be validating those?

From what you're saying in the docstring of `_record_user_external_id_txn` it doesn't look like we can realistically validate these? Or am I misunderstanding things?

> Should we update the markdown documentation?
> Should we update the documentation for the module API?

It could be nice having a note saying that no error will be raised if the provider ID isn't valid so users and modules don't work with false assumptions.

**Comment by clokep** [comment]
> > Does this sound fine or should we really be validating those?
> 
> From what you're saying in the docstring of `_record_user_external_id_txn` it doesn't look like we can realistically validate these? Or am I misunderstanding things?

We could check them against the current config, but yeah pretty much there are situations you actually want it to differ frmo the current config.

> > Should we update the markdown documentation?
> > Should we update the documentation for the module API?
> 
> It could be nice having a note saying that no error will be raised if the provider ID isn't valid so users and modules don't work with false assumptions.

Will do! 👍

**Pull request merged**